File: src/RoboTasker.Api/Models/Robots/UpdateRobotRequest.cs
using RoboTasker.Application.Robots.Robots.CreateRobot;
using RoboTasker.Application.Robots.Robots.UpdateRobot;

namespace RoboTasker.Api.Models.Robots;

public class UpdateRobotRequest
{
    public string? Name { get; set; }
    public Guid? CategoryId { get; set; }
    public IList<UpdateRobotCommandPropertyValue>? UpdatedProperties { get; set; }
    public IList<Guid>? DeletedCustomProperties { get; set; }
    public IList<CreateRobotCommandCustomPropertyItem>? NewCustomProperties { get; set; }
    public IList<UpdateRobotCapabilityItem>? DeletedCapabilities { get; set; }
    public IList<UpdateRobotCapabilityItem>? NewCapabilities { get; set; }
}

File: src/Nerbotix.UI/src/app/models/robots/robots/requests/update-robot-request.ts
import {Guid} from 'guid-typescript';
import {
  CreateRobotCapabilityRequest,
  CreateRobotCustomPropertyRequest,
  CreateRobotHttpCommunicationRequest, CreateRobotMqttCommunicationRequest
} from './create-robot-request';
import {CommunicationType} from '../../../../enums/communication-type.enum';

export interface UpdateRobotRequest {
  name?: string | undefined;
  categoryId?: string | undefined;
  updatedProperties?: UpdateRobotPropertyRequest[] | undefined;
  deletedCustomProperties?: Guid[] | undefined;
  newCustomProperties?: CreateRobotCustomPropertyRequest[] | undefined;
  deletedCapabilities?: CreateRobotCapabilityRequest[] | undefined;
  newCapabilities?: CreateRobotCapabilityRequest[] | undefined;
  communicationType?: CommunicationType;
  httpCommunication?: CreateRobotHttpCommunicationRequest;
  mqttCommunication?: CreateRobotMqttCommunicationRequest;
}

interface UpdateRobotPropertyRequest {
  propertyId: Guid;
  value: any;
}
